From 595e0877cca0d1b9a65e5c59cdb9651971154a5b Mon Sep 17 00:00:00 2001
Message-Id: <595e0877cca0d1b9a65e5c59cdb9651971154a5b.1687302523.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1687302523.git.sharlatanus@gmail.com>
References: <cover.1687302523.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Tue, 20 Jun 2023 21:09:24 +0100
Subject: [PATCH 14/20] gnu: python-pyerfa: Fix tests.

* gnu/packages/astronomy.scm (python-pyerfa): Fix tests.
[snippet]: Use G-expressions.
[arguments]: Use G-expressions. Add new phase 'tests-preparation.
---
 gnu/packages/astronomy.scm | 26 ++++++++++++++++----------
 1 file changed, 16 insertions(+), 10 deletions(-)

diff --git a/gnu/packages/astronomy.scm b/gnu/packages/astronomy.scm
index 829ee6bcd4..514aa53258 100644
--- a/gnu/packages/astronomy.scm
+++ b/gnu/packages/astronomy.scm
@@ -3037,17 +3037,23 @@ (define-public python-pyerfa
         (base32 "0f8zykzxjsiwv5ibdn5asla2ng2xl0xdkrcrrd61j31mb3xbnzyp"))
        (modules '((guix build utils)))
        (snippet
-        '(begin
-           ;; Remove bundled submodule library.
-           (delete-file-recursively "liberfa")
-           #t))))
-    (build-system python-build-system)
+        #~(begin
+            ;; Remove bundled submodule library.
+            (delete-file-recursively "liberfa")))))
+    (build-system pyproject-build-system)
     (arguments
-     `(#:phases
-       (modify-phases %standard-phases
-         (add-before 'build 'use-system-liberfa
-           (lambda _
-             (setenv "PYERFA_USE_SYSTEM_LIBERFA" "1"))))))
+     (list
+      ;; Disable only one failing test:
+      ;; AttributeError: __warningregistry__
+      #:test-flags #~(list "-k" "not test_errwarn_reporting")
+      #:phases
+      #~(modify-phases %standard-phases
+          (add-before 'build 'use-system-liberfa
+            (lambda _
+              (setenv "PYERFA_USE_SYSTEM_LIBERFA" "1")))
+          (add-before 'check 'tests-preparation
+            (lambda _
+              (invoke "python" "setup.py" "build_ext" "--inplace"))))))
     (native-inputs
      (list python-pytest-doctestplus python-pytest python-setuptools-scm))
     (inputs
-- 
2.40.1

